src/test/groovy/com/vacation_bot/vacation/VacationIntegrationTest.groovy

```diff
@@ -0,0 +1,126 @@
+package com.vacation_bot.vacation
+
+import com.vacation_bot.AbstractSpockIntegrationTest
+import com.vacation_bot.domain.models.UserModel
+import com.vacation_bot.domain.models.VacationRequestBody
+import com.vacation_bot.domain.models.VacationTotalModel
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+
+/**
+ * Integration level test of {@Link VacationGateway}.
+ */
+class VacationIntegrationTest extends AbstractSpockIntegrationTest{
```

Comment:
Each test should test 4 cases with the help of `where` block:
1. successful reservation. user still have vacation
2. successful reservation. vacation is exhausted
3. failed reservation. user still have vacation
4. failed reservation. vacation is exhausted

You can change startDate, endDate and vacationTotalDays values in the where block.